src/systemctl/systemctl.c

```diff
@@ -3548,6 +3548,22 @@ static int load_kexec_kernel(void) {
         return 0;
 }
 
+static int kill_user_manager(void) {
+        int r;
+        char *pid_env;
+        pid_t pid;
+
+        pid_env = getenv("MANAGERPID");
```

Comment:
Is there a better way that doesn't depend on $MANAGERPID being present?

Reply:
Doesn't really look like this pid is available anywhere else that's easily obtainable... One thing that occurred me was, when the environment variable is set, also set a property of the user manager such as ManagerPid= or Pid= and then query it later... But perhaps using $MANAGERPID from the environment is good enough? Thoughts?